### DISCUSSION TITLE
关于命令匹配的一些疑问

### DISCUSSION THREAD

**Discussion by Freeman-fan** [nonebot/discussions#507] [问答] [answered]
1.能不能让命令和参数之间不需要空格区分，例如“.echo你好"=".echo 你好"。
2.如何匹配@bot的消息，如"@bot echo 你好"。根据我的测试，@消息在raw_message中表现为"[CQ:at,qq=123456]" （这个值来源于一条用on_command匹配的消息，它在消息的后半段有@bot）。那么如果我希望匹配以@bot开头的消息，需要怎么做。
3.如上面所说，有无方法能不使用预设的匹配，将所有信息都进入到插件，由插件进行处理。如果这个能实现，那么上述两点我也能够自行完成匹配。

以上，十分感谢

**Comment by yanyongyu** [comment, marked as answer]
1. 本来就不需要空格啊，如果你是想要强制要空格的话，可以使用 `force_whitespace` 参数
2. nb 内置 rule to_me 就是识别是否 at 机器人或者私聊的规则
3. 你需要的就是 on_message

以上问题在文档中均有涉及

**Reply by Freeman-fan** [reply]
关于`force_whitespace`和rule `to_me`，我翻阅nb1的文档似乎没有找到，但是在nb2的文档中找到了对应的内容。
我使用的是nb1，请问有对应的规则吗。

**Reply by yanyongyu** [reply]
nb1 已经有点久远了不太记得了，我印象中似乎有点麻烦，建议还是使用 nb2
